### PR TITLE
update Go to `1.20.6`

### PR DESCRIPTION
**Pull request by 0xmohit**
## Description
Update Go to version 1.20.2

## Related Issue(s)

Fixes #

## Release Notes

```release-note
```

### CODE REVIEW

**Comment by kylos101** [comment]
:wave: @atduarte @loujaybee , for your :eyes: , I'm not sure when you'll be considering the next changelog, but this is certainly a candidate to give users a heads up about.

**Comment by Siddhant-K-code** [comment]
Will followup on this, after release of https://github.com/gitpod-io/workspace-images/pull/1017

**Comment by PatrickRice-KSC** [comment]
Hey @Siddhant-K-code  - any update on this one? We use gitpod on an open source project that uses go 1.20, and it's really obnoxious right now to tell our contributors to manually install it. We could add a `task` that installs it, but we'd rather just use the workspace image.

**Comment by axonasif** [comment]
> but we'd rather just use the workspace image.

👋 @PatrickRice-KSC, you can do exactly that through a custom workspace image. Please see:
- https://www.gitpod.io/docs/introduction/languages/go#setting-up-a-custom-dockerfile

You can change the `FROM gitpod/workspace-base:latest` line to `FROM gitpod/workspace-base:2023-03-24-22-45-37` and `ENV GO_VERSION=1.17` to `ENV GO_VERSION=1.20.2` from the guide above for latest ubuntu jammy and go versions.

**Comment by kylos101** [comment]
@loujaybee @laushinka is this something IDE team can consider for the future?

File: chunks/lang-go/chunk.yaml
Comment (Siddhant-K-code) [inline chunks/lang-go/chunk.yaml:7]: Recent stable version is `1.20.3` now. ([reference](https://go.dev/dl/#:~:text=e12384311403f1389d14cc1c1295bfb4e0dd5ab919403b80da429f671a223507-,go1.20.3.linux%2Damd64.tar.gz,-Archive))

Comment (gengjiawen) [reply chunks/lang-go/chunk.yaml:7]: 1.20.4 now

**Comment by kylos101** [comment]
@iQQBot :wave: , for example, if we merge this today, it would be good to share a related changelog entry, so that next week, when this lands as latest, folks understand their options. Let me know if you need anything else?

**Comment by gengjiawen** [comment]
> @iQQBot 👋 , for example, if we merge this today, it would be good to share a related changelog entry, so that next week, when this lands as latest, folks understand their options. Let me know if you need anything else?

put this in https://github.com/gitpod-io/workspace-images/blob/main/CHANGELOG.md ?

**Comment by kylos101** [comment]
> > @iQQBot wave , for example, if we merge this today, it would be good to share a related changelog entry, so that next week, when this lands as latest, folks understand their options. Let me know if you need anything else?
> 
> put this in https://github.com/gitpod-io/workspace-images/blob/main/CHANGELOG.md ?

@loujaybee @iQQBot @filiptronicek I'd recommend a changelog entry on the website. The users are options now are to do nothing (use latest), use the present tag that is latest now (to avoid the change), or use the tag associated with this to test the change, before it is usable as latest. Similar to [this](https://www.gitpod.io/changelog/workpace-images-os-update-and-breaking-changes) entry.

**Comment by gengjiawen** [comment]
> I'd recommend a changelog entry on the website. 

tbh this will bring to much extra work for contributor, maybe put on `CHANGELOG.md` first, send weekly update to website by maintainer.

**Comment by akosyakov** [comment]
I wonder why don't have the following process:
- release `latest` once a month
- nightly publish `next` for people to try and provide feedback
- merge PRs without being afraid because of above, stop merging a week or so before `latest` release

Run it all automatically as we do with VS Code, JetBrains, let the community help catch issues.

**Review by kylos101 — APPROVED** [review]
@filiptronicek @iQQBot I like the suggestion from @gengjiawen , about using the changelog.md file in this repo to share related chnages, it reduces friction. Approving for now, to unblock.

**Review by filiptronicek — APPROVED** [review]

**Comment by filiptronicek** [comment]
Tested independently in a workspace

File: chunks/lang-go/chunk.yaml
Comment (kylos101) [inline chunks/lang-go/chunk.yaml:5]: Good call, @filiptronicek !! :tada:

**Pull request merged**
